### PR TITLE
#39 Use multipart form data when starting an import job.

### LINKED ISSUE #39: [Bulk Import] Convert to use multipart form-data
The import service now requires multi part form data instead of JSON. 

> Use multipart/form-data content type to support upload of custom import.js https://github.com/adobe/spacecat-api-service/pull/487

The `startButton` click handler will need to be rewritten to create a form data object.

```
  const formData = new FormData();
  formData.append('urls', JSON.stringify(urlsArray));
  formData.append('options', JSON.stringify(options));
  formData.append('customHeaders', fields.headers.value);
  formData.append('importScript', fields.importScript.files[0]);

  const newJob = await service.startJob(formData);
```

The start service will then also need to be updated to accept a form data object.

```
      const headers = new Headers(this.#getAuthHeaders());
      headers.delete('Content-Type');
    const resp = await fetch(`${this.endpoint}${IMPORT_JOBS_PATH}`, {
      method: 'POST',
      headers,
      body: formData,
    });
```

### PR DESCRIPTION
**Pull request by atopper**
Fix #39 

Test URLs:
- Before: https://main--helix-labs-website--adobe.aem.live/tools/import/index.html
- After: https://import-formdata--helix-labs-website--adobe.aem.live/tools/import/index.html

### CODE REVIEW

File: tools/import/import.js [hunk 1]
```diff
@@ -196,25 +185,36 @@ function addJobsList(jobs) {
...
+      }
     }
     const options = {
       ...buildOptions(form),
       pageLoadTimeout: parseInt(fields.timeout.value || '100', 10),
-      customHeaders,
     };
-    const importScript = await getImportScript(fields.importScript);
-    const newJob = await service.startJob({ urls: urlsArray, options, importScript });
+    const importScript = fields.importScript.files[0] ?? undefined;
```
Comment (atopper) [inline tools/import/import.js:205]: Is this sufficient for what the service requires? ☝️  (`...files[0]`).

File: tools/import/importservice.js [hunk 2]
```diff
@@ -118,16 +118,35 @@ export default class ImportService {
...
-    const body = importScript ? { urls, options, importScript } : { urls, options };
     try {
+      const formData = new FormData();
+      formData.append('urls', JSON.stringify(urls));
+      formData.append('options', JSON.stringify(options));
+      formData.append('customHeaders', customHeaders);
+      formData.append('importScript', importScript);
+
+      const authHeaders = this.#getAuthHeaders();
+      delete authHeaders['Content-Type'];
```
Comment (arumsey) [inline tools/import/importservice.js:144]: I am still seeing application/json in the request when using this approach.

File: tools/import/importservice.js [hunk 3]
```diff
@@ -118,16 +118,35 @@ export default class ImportService {
...
+
       const resp = await fetch(`${this.endpoint}${IMPORT_JOBS_PATH}`, {
         method: 'POST',
-        headers: this.#getAuthHeaders(),
-        body: JSON.stringify(body),
+        headers: authHeaders,
+        body: formData,
       });
       if (resp.ok) {
```
Comment (arumsey) [inline tools/import/importservice.js:151]: When resp is not ok we should report on the X-Error and x-invocation-id header values.

**Review by arumsey — APPROVED** [review]

File: tools/import/importservice.js (hunk 2 above)
Comment (atopper) [reply tools/import/importservice.js:149]: Bruce and I saw it in the Response header, but not in the Request header which defaulted correctly.

**Pull request merged**